Doc/whatsnew/3.11.rst

```diff
@@ -1927,6 +1927,15 @@ Porting to Python 3.11
   paths and then modify them, finish initialization and use :c:func:`PySys_GetObject`
   to retrieve :data:`sys.path` as a Python list object and modify it directly.
 
+* CPython no longer assumes that a ``__dict__`` exists after an object. C extension
+  types with inheritance and ``__dict__`` may have MRO problems. To fix this, allow
+  CPython to manage the ``__dict__`` of instances of your type by setting
+  :const:`Py_TPFLAGS_MANAGED_DICT` in :c:member:`~PyTypeObject.tp_flags`.
```

Comment:
This really raises more questions than it answers. In what cases did CPython assume a ``__dict__`` exists after an object? (Didn't it put it there itself before 3.11?) What does it mean for a type to be “with inheritance”? What kind of MRO problems can you fix with the flag?